src/protocol.rs

```diff
@@ -262,15 +275,14 @@ impl SignatureShare {
         }?;
 
         // Keep the same r, add in the s value
-        Ok(Self {
-            r: Some(r),
-            s: self.s + share.s,
-        })
+        Ok(Self::new(Some(r), self.s + share.s))
     }
 
-    /// Converts the [`SignatureShare`] into a signature.
+    /// Convert the [`SignatureShare`] into an ECDSA signature.
+    // XXX we should check here that the signature is valid! (As is done in the
+    // paper.)
```

Comment:
Actually verify the signature. I guess my concern is, at what point do we know if the signature is valid or malformed? Suppose we don't do this signature check, and the caller just assumes things are valid, passes on or stores the signature, and only later runs the `sig.verify` call. The failure there could either be (1) the signature was indeed invalid for the particular digest passed to `verify`, or (2) the signature was invalidly constructed (due to e.g. a bogus share). At the point of the `sig.verify` call the caller wouldn't know which, whereas at this point of the code, if we verify the signature, we can rule out #2 above.

Now, maybe the cost-benefit analysis isn't worth it: we _are_ paying the cost of verifying the signature and extra time, and without the ability to identify the culprit this extra verification might not be worth it. But in the worst case we should document this. I'm happy to do that as part of this issue if we want to avoid that extra cost.

Reply:
Are you saying that problem 2 is the case where you get a bunch of signature shares and some of them have different `r` values, while 1 is, like, the σ / s values combine to something that isn't a signature on the digest? or, say, you asked for a signature on `x` and then it fails to verify for `x'`, but you can't tell if that's because `r` was borked or `x'` was wrong?

I'm not really worried about the cost of verifying the signature, mostly just trying to think through our use case and what inputs the caller of this method will have access to. You've moved these steps to the `into_signature` function; I think it would be fine to pass a message and public key to that method, but I'm also not convinced that checking it at the time of signature creation is significantly better than checking it in a separate verification function.

Reply:
My reading of Figure 8 is that, once we reconstruct σ, we validate the signature so that if it fails we can run the "blame" protocol (Steps (a)-(c)). If we _don't_ check the validity of the signature here, we can't necessarily run the "blame" protocol later (since the consumer of the signature isn't necessarily the one participating in the protocol).

Now, I realize we don't implement the "blame" protocol currently, so maybe it doesn't matter, but my main point was that checking the validity here could be used to at least identify that the signature was constructed incorrectly, whereas checking it later could either mean that the input message was wrong (i.e., we checked the signature on the wrong message), _or_ the signature was constructed incorrectly.

Does that help clarify things?

I'm okay forgoing the verification check here, but if so maybe it's worth a note saying something similar to the above?

Reply:
I talked to @indomitableSwan about this and I think the best way forward is to plan to update the library to actually implement the whole signing protocol as written, rather than change it like this. This would involve making a new `ProtocolParticipant` for signing and adding the one round of communication in there.

In lieu of that, I think improving documentation here (this deviates from the protocol and by not checking signature validity we lose identifiable abort) is an appropriate scope for this PR.

Reply:
I advocate for including the verification step. 

We can separately consider whether or not we want to support signing _without_ this verification. But at least so far, our intended use follows the protocol here (including leveraging identifiable abort properties).

Reply:
Yes, that looks good. Thanks!